lib/dal/src/cached_module.rs

```diff
@@ -288,271 +349,171 @@ impl CachedModule {
-        let query = "
-            SELECT DISTINCT ON (schema_id)
-                id,
-                schema_id,
-                schema_name,
-                display_name,
-                category,
-                link,
-                color,
-                description,
-                component_type,
-                latest_hash,
-                created_at,
-                NULL::bytea AS package_data,
-                scoped_to_user_pk
-            FROM cached_modules
-            WHERE scoped_to_user_pk IS NULL
-            ORDER BY schema_id, created_at DESC
-        ";
-
-        let rows = ctx.txns().await?.pg().query(query, &[]).await?;
-
-        let mut result = vec![];
-
-        for row in rows {
-            result.push(row.try_into()?);
-        }
+        let query = format!(
+            "
+                SELECT DISTINCT ON (schema_id)
+                    {CACHED_MODULE_LIST_FIELDS}
+                FROM cached_modules
+                WHERE scoped_to_user_pk IS NULL
+                ORDER BY schema_id, created_at DESC
+            "
+        );
 
-        Ok(result)
+        let rows = ctx.txns().await?.pg().query(&query, &[]).await?;
+        rows.into_iter().map(TryInto::try_into).try_collect()
     }
 
     pub async fn latest_modules(ctx: &DalContext) -> CachedModuleResult<Vec<CachedModule>> {
         let HistoryActor::User(user_pk) = ctx.history_actor() else {
             return Self::latest_user_independent_modules(ctx).await;
         };
 
-        let query = "
-            SELECT DISTINCT ON (schema_id)
-                id,
-                schema_id,
-                schema_name,
-                display_name,
-                category,
-                link,
-                color,
-                description,
-                component_type,
-                latest_hash,
-                created_at,
-                NULL::bytea AS package_data,
-                scoped_to_user_pk
-            FROM cached_modules
-            WHERE scoped_to_user_pk IS NULL OR scoped_to_user_pk = $1
-            ORDER BY schema_id, created_at DESC
-        ";
-
-        let rows = ctx.txns().await?.pg().query(query, &[&user_pk]).await?;
-
-        let mut result = vec![];
-
-        for row in rows {
-            result.push(row.try_into()?);
-        }
+        let query = format!(
+            "
+                SELECT DISTINCT ON (schema_id)
+                    {CACHED_MODULE_LIST_FIELDS}
+                FROM cached_modules
+                WHERE scoped_to_user_pk IS NULL OR scoped_to_user_pk = $1
+                ORDER BY schema_id, created_at DESC
+            "
+        );
 
-        Ok(result)
+        let rows = ctx.txns().await?.pg().query(&query, &[&user_pk]).await?;
+        rows.into_iter().map(TryInto::try_into).try_collect()
     }
 
-    // TO-DO: Paul
-    // I will refactor this + insert together - this is to temporarily get something working
-    async fn insert_private_module(
+    async fn insert(
```

Comment:
Literally everything except scoped_to_user_pk was shared between this and insert_private_module, so it's down to a single method that gets passed an optional scoped_to_user_pk.